### ISSUE TITLE
Support for pushing multiple tags other than 'latest'

### ISSUE THREAD

**Issue by dackroyd** [concourse/docker-image-resource#98]
I am looking to build and push docker images into multiple tags, none of which are 'latest'. This is to be able to push tags like `onbuild` and `onbuild-${githash}`, and then be able to use these by either taking the latest `onbuild`, or pinning the version by using the hash.

With regular images, I'm able to push both `latest` and `${githash}` tags from the same build thanks to `tag_as_latest`, but this is not possible when not using the `latest` tag for one of those (without building twice)

**Comment by dackroyd** [comment]
I have since worked around this by adding a subsequent job in the pipeline which loads the previously pushed image then pushes it with a different tag:

```
- name: build
  plan:
  - get: repo
    trigger: true
  - task: write-tag-files
     ... generate tag files ...
  - put: docker-repo
    params:
      build: repo
      dockerfile: repo/Dockerfile
      tag: tags/onbuild-version

- name: push-other-tags
  plan:
  - aggregate:
      - get: repo
        passed: [build]
        trigger: true
      - get: docker-repo
        passed: [build]
        params: {save: true}
  - task: write-tag-files
     ... generate tag files again (since I didn't push the files anywhere) ...
  - put: docker-repo
    params:
      load: docker-repo
      tag: tags/onbuild
```

**Comment by gaelL** [comment]
Docker allow to specify several `-t` parameters during a build https://github.com/moby/moby/issues/863

An easy way to implement it would be to simply add a generic parameter like `docker_args` here 
 https://github.com/concourse/docker-image-resource/blob/81bd23138dfd92b9699d829b3868c3f1657ed953/assets/out#L133

That would allow to specify specific docker args if needed and also several tags for the build.

**Comment by chendrix** [comment]
Because of @gaelL's findings, I'm going to close this issue!

Thanks again all for your findings.

**Comment by gaelL** [comment]
There is a little misunderstanding, as far as I understand we can give `build_args` 
https://github.com/concourse/docker-image-resource/blob/81bd23138dfd92b9699d829b3868c3f1657ed953/assets/out#L126

which correspond to docker `--build-arg` only

eg:

```
echo '{"foo": "bar", "bar": "foo"}'|jq -r 'with_entries(.key |= "--build-arg " + . )|with_entries(.key = .key + "=" +.value)|keys|join(" ")'
--build-arg bar=foo --build-arg foo=bar
```

Looking https://github.com/concourse/docker-image-resource/blob/81bd23138dfd92b9699d829b3868c3f1657ed953/assets/out#L133 in the docker command 
 `docker build -t "${repository}:${tag_name}" $expanded_build_args -f "$dockerfile" $cache_from "$build"` the variable `expanded_build_args` come from `build_args` or `build_args_file` **BUT** the content is parsed and prefixed by `--build-arg`.

So in the current state we can't add additional args to docker. This is why I was thinking of adding a new parameter like `$additional_parameters` allowing us to directly give docker arguments (or just rework `expanded_build_args` to be less strict) : 

```
- docker build -t "${repository}:${tag_name}" $expanded_build_args -f "$dockerfile" $cache_from "$build"
+ docker build -t "${repository}:${tag_name}" $expanded_build_args $additional_parameters -f "$dockerfile" $cache_from "$build"
```

**Comment by chendrix** [comment]
Ah sorry I didn't see the part where the build args are parsed and prefixed with --build

**Comment by chendrix** [comment]
This sounds like a story that's straightforward enough to be a PR from a community member.

I'm going to attach this to the milestone "Help Wanted"

**Comment by philicious** [comment]
a way of adding/pushing more than one tag besides `latest` would be great !

**Comment by zhenyulin** [comment]
+1

**Comment by jtarchie** [comment]
Would this PR (https://github.com/concourse/docker-image-resource/pull/124) be useful?

**Comment by dackroyd** [comment]
@jtarchie looks like that fits the requirements here

**Comment by tm-guimaraes** [comment]
Apparently there's #171 for this.
This feature is nice for some workflows.

**Comment by ghostsquad** [comment]
Can this issue be closed now that #171 is merged?

**Comment by philicious** [comment]
@ghostsquad yes
